File: ADCP_Nortek_Aquadopp_AWAC_Vector/Andere talen & old/Matlab/script_mariakerke_2015_09.m
% SCRIPT_MARIAKERKE_2015_09 is a script that generates the figures of the
% Sept-Nov 2015 Mariakerke campaign (HERCULESI:1 aquadopp and 1 AWAC)

% Required packages: suplabel
%
% Developed at Flanders Hydraulics Research (FHR), Antwerp, Belgium
% Author: Olivier Gourgue

% test changement

close all
clear all

% dpl_param.data_treatment = 'storm';
out_param.save_format = 'png';
dpl_param.data_treatment = 'storm';


% HERCULESI awac
% dpl_param.data_treatment = 'storm';
out_param.save_format = 'png';
dpl_param.data_treatment = 'storm';


dpl_param.folder_name = 'F:\MyMatlab\CampaignAnalysis\nortek_matlab\data\2015_09_mariakerke\M1_Hercules\AWAC\Processed23092015_14112015';
dpl_param.dpl_name = 'Herculesa660602_23092015_14112015';
dpl_param.z0 = -6.5; %depth -6.5m TAW
dpl_param.reference_height = 'awac';
dpl_param.vertical_dir = 'upward';
dpl_param.time_gap = -14 ; % to change -14s (instrument was earlier than the computer)
dpl = load_deployment_files(dpl_param);

out_param.time_start = datenum('23-Sept-2015 00:01:01');
out_param.time_stop = datenum('14-Nov-2015 00:01:01');
%out_param.profile_days_per_subplot = 3;
out = read_output_parameters(dpl, out_param);

figure_ts_instrument_orientation(dpl, out);
figure_ts_depth_averaged_velocity_components(dpl, out);
figure_ts_velocity_norm_profile(dpl, out, 7, NaN, NaN, NaN);
figure_depth_averaged_horizontal_velocity(dpl, out, 0, NaN, [NaN NaN NaN NaN]);

File: ADCP_Nortek_Aquadopp_AWAC_Vector/Andere talen & old/Matlab/script_mariakerke_2015_09.py
from __future__ import annotations

import numpy as np

from campaign_runner import run_campaign_entries, show


def main() -> list[dict]:
    entries = [
        {
            "dpl_param": {
                "folder_name": "F:/MyMatlab/CampaignAnalysis/nortek_matlab/data/2015_09_mariakerke/M1_Hercules/AWAC/Processed23092015_14112015",
                "dpl_name": "Herculesa660602_23092015_14112015",
                "z0": -6.5,
                "reference_height": "awac",
                "vertical_dir": "upward",
                "time_gap": -14,
                "data_treatment": "storm",
            },
            "out_param": {
                "save_format": "png",
                "time_start": "23-Sep-2015 00:01:01",
                "time_stop": "14-Nov-2015 00:01:01",
            },
            "days_per_subplot": 7,
            "ebb_flood": False,
            "ellipse_bounding_box_display": [np.nan, np.nan, np.nan, np.nan],
        }
    ]
    return run_campaign_entries(entries)


if __name__ == "__main__":
    main()
    show()
